libs/api-bot/src/Network/Wire/Bot/Monad.hs

```diff
@@ -447,10 +464,11 @@ withCachedBot t f = do
 -- Assertions
 
 data EventAssertion = EventAssertion
-    { _assertType :: !EventType
-    , _assertTime :: !UTCTime
-    , _assertPred :: Event -> Bool
-    , _assertOut  :: !(Maybe (TMVar (Maybe Event)))
+    { _assertType  :: !EventType
+    , _assertTime  :: !UTCTime
+    , _assertPred  :: Event -> Bool
+    , _assertOut   :: !(Maybe (TMVar (Maybe Event)))
+    , _assertStack :: !CallStack
```

Comment:
Added the `_assertStack` field for recording the call stack at the time of scheduling the assert